pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -643,12 +643,12 @@ def _save_last_checkpoint(self, trainer: "pl.Trainer", monitor_candidates: Dict[
             return
 
         filepath = self.format_checkpoint_name(monitor_candidates, self.CHECKPOINT_NAME_LAST)
-        trainer.save_checkpoint(filepath, self.save_weights_only)
 
         if self.last_model_path and self.last_model_path != filepath:
             trainer.strategy.remove_checkpoint(self.last_model_path)
 
         self.last_model_path = filepath
+        trainer.save_checkpoint(filepath, self.save_weights_only)
```

Comment:
The problem of putting this here is that if `save_checkpoint` fails (maybe the user has a bug in a `on_save_checkpoint` hook) then training stops and the previous checkpoint is gone.

Reply:
Agreed, that is true but if we don't do it here, as said it's losing the previous last.ckpt state while resuming the run. 
Do you have any other suggestion?

Reply:
or what might be causing this issue?

Reply:
Followed up in the issue